lib/errors.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+/**
+ * Code value constants
+ */
+
+var ERR_MOCHA_NO_FILES_MATCH_PATTERN = 'ERR_MOCHA_NO_FILES_MATCH_PATTERN';
+var ERR_MOCHA_MISSING_CALLBACK = 'ERR_MOCHA_MISSING_CALLBACK';
+var ERR_MOCHA_INVALID_REPORTER = 'ERR_MOCHA_INVALID_REPORTER';
+var ERR_MOCHA_INVALID_INTERFACE = 'ERR_MOCHA_INVALID_INTERFACE';
+var ERR_MOCHA_NOT_SUPPORTED = 'ERR_MOCHA_NOT_SUPPORTED';
+var ERR_MOCHA_INVALID_ARG_VALUE = 'ERR_MOCHA_INVALID_ARG_VALUE';
+var ERR_MOCHA_MISSING_ARGUMENT = 'ERR_MOCHA_MISSING_ARGUMENT';
+var ERR_MOCHA_UNDEFINED_ERROR = 'ERR_MOCHA_UNDEFINED_ERROR';
+
+module.exports.ERR_MOCHA_NO_FILES_MATCH_PATTERN = ERR_MOCHA_NO_FILES_MATCH_PATTERN;
+module.exports.ERR_MOCHA_MISSING_CALLBACK = ERR_MOCHA_MISSING_CALLBACK;
+module.exports.ERR_MOCHA_INVALID_REPORTER = ERR_MOCHA_INVALID_REPORTER;
+module.exports.ERR_MOCHA_INVALID_INTERFACE = ERR_MOCHA_INVALID_INTERFACE;
+module.exports.ERR_MOCHA_NOT_SUPPORTED = ERR_MOCHA_NOT_SUPPORTED;
+module.exports.ERR_MOCHA_INVALID_ARG_VALUE = ERR_MOCHA_INVALID_ARG_VALUE;
+module.exports.ERR_MOCHA_MISSING_ARGUMENT = ERR_MOCHA_MISSING_ARGUMENT;
+module.exports.ERR_MOCHA_UNDEFINED_ERROR = ERR_MOCHA_UNDEFINED_ERROR;
+
+/**
+ * Custom Errors objects
+ */
+
```

Comment:
As question for all concerned, should design be to have everything be completely independent or to subclass from primary? Could see advantage for user in having latter capability being able to test caught error as Mocha-related or not.

Reply:
I thought this only applied if we utilised custom error classes (for MochaError), something we are now not doing (see comment https://github.com/mochajs/mocha/pull/3467#issuecomment-445924038) . Let me know if you have any further thoughts on this.

Reply:
I thought this only applied if we utilised custom error classes (for MochaError), something we are now not doing (see comment https://github.com/mochajs/mocha/pull/3467#issuecomment-445924038) . Let me know if you have any further thoughts on this.

Reply:
No further thoughts. There was no comment why at the time.